rocAL/rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -150,17 +184,14 @@ rocalJpegFileSourceSingleShard(
                                evaluate_image_data_set(decode_size_policy, StorageType::FILE_SYSTEM, DecoderType::TURBO_JPEG,
                                                        source_path, "");
         auto [color_format, num_of_planes] = convert_color_format(rocal_color_format);
-
-
-
         INFO("Internal buffer size width = "+ TOSTR(width)+ " height = "+ TOSTR(height) + " depth = "+ TOSTR(num_of_planes))
 
-        std::vector<size_t> dims = {context->user_batch_size(), height, width, num_of_planes};
+        auto [tensor_layout, dims] = convert_color_format_to_layout(rocal_color_format, context->user_batch_size(), height, width, num_of_planes);
```

Comment:
move the implementation of this function to tensor class

Reply:
These changes were a part of the previous PR and these changes were visible here as it was not merged before.
This file is not a part of this PR. Will be changed in the upcoming PR for readers and meta nodes.